pkg/apm/applications.go

```diff
@@ -88,6 +73,54 @@ func (apm *APM) DeleteApplication(applicationID int) (*Application, error) {
 	return &response.Application, nil
 }
 
+// MetricNames is used to retrieve a list of known metrics and their value names for the given resource.
+//
+// https://rpm.newrelic.com/api/explore/applications/metric_names
+func (apm *APM) MetricNames(applicationID int, params MetricNamesParams) ([]*MetricName, error) {
```

Comment:
Can we name this `GetMetricNames()`? We've been prefixing the "action type" to our client methods - i.e. Create*, List*, Get*, Update*, Delete*

Reply:
Sounds reasonable to me.  I can do the same for the `MetricData`.